### ISSUE TITLE
unsupported media type for "application/json; charset=utf-8"

### ISSUE THREAD

**Issue by mk811** [cdimascio/express-openapi-validator#124]
Hi, my company has a rule that we mandatory set `Content-Type: application.json; charset=utf-8` to all the request/response header. (I know the `charset=utf-8` is probably redundant for json though.)
Since OpenAPI Validator drop charactors of Content-Type after the charactor of `;` as below, our OpenAPI Validator cannot find the appropriate request body schema and respond `unsupported media type application/json`.

https://github.com/cdimascio/express-openapi-validator/blob/04a84e513ab253f7d7e655c53bdac7978ba43093/src/middlewares/util.ts#L6-L17

Sample api spec is described as below. Due to the issue above, requests which has `Content-Type: application.json; charset=utf-8` cannot be associated with the schema `$ref: '#/components/schemas/user'`.

```yaml
paths:
  /user/:userId:
    post:
# --- skip ---
      requestBody:
        content:
          application/json; charset=utf-8:
            schema:
              $ref: '#/components/schemas/user'
# --- skip ---
```

I think we should use original raw value of `Content-Type` in request header to find appropriate schema.

https://github.com/cdimascio/express-openapi-validator/blob/04a84e513ab253f7d7e655c53bdac7978ba43093/src/middlewares/openapi.request.validator.ts#L196 

If you feel okay with my suggestion, let me create PR.

**Comment by cdimascio** [comment]
@mk811 i agree, the raw should be used. a PR is will be greatly appreciated!
we likely also want to fall back to the `media-type` if the `charset` or `boundary` are not present

The syntax is as such:
e.g.
```
Content-Type: text/html; charset=UTF-8
Content-Type: multipart/form-data; boundary=something
```
we should fall back to the `media-type` if charset or boundary are missing

**Comment by mk811** [comment]
@cdimascio  Sounds great. Thank you for the quick reply.

To sum up, if the `Content-Type` has `charset` or `boundary` in itself, we will use `Content-Type` to find appropriate request/response body from spec yaml, e.g. `text/html; charset=UTF-8`.
If not, we will use `media-type` instead, e.g. `text/html`.

Does it sound good for you?

**Comment by cdimascio** [comment]
Yes, that sounds good.

**Comment by cdimascio** [comment]
@mk811 how are you making out? If u have any questions or want to discuss more freely, feel free to join us on gitter. https://gitter.im/cdimascio-oss/community

**Comment by cdimascio** [comment]
fix available in `v2.15.1`
